<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Интерполяция сплайнами</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(99, 99, 99);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="inputGroup">
      <property name="styleSheet">
       <string notr="true">color: rgb(170, 170, 255);</string>
      </property>
      <property name="title">
       <string>Параметры интерполяции</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="styleSheet">
            <string notr="true">color: rgb(170, 170, 255);</string>
           </property>
           <property name="text">
            <string>Функция f(x):</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="functionInput">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(206, 168, 218);
color: rgb(0, 0, 0);</string>
           </property>
           <property name="text">
            <string></string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="styleSheet">
            <string notr="true">color: rgb(170, 170, 255);</string>
           </property>
           <property name="text">
            <string>Начало отрезка a:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="aInput">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(206, 168, 218);
color: rgb(0, 0, 0);</string>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="value">
            <double>-1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="styleSheet">
            <string notr="true">color: rgb(170, 170, 255);</string>
           </property>
           <property name="text">
            <string>Конец отрезка b:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="bInput">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(206, 168, 218);
color: rgb(0, 0, 0);</string>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>-1000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="styleSheet">
            <string notr="true">color: rgb(170, 170, 255);</string>
           </property>
           <property name="text">
            <string>Количество узлов n:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="nInput">
           <property name="styleSheet">
            <string notr="true">background-color: rgb(206, 168, 218);
color: rgb(0, 0, 0);</string>
           </property>
           <property name="minimum">
            <number>3</number>
           </property>
           <property name="maximum">
            <number>200</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="calculateBtn">
         <property name="styleSheet">
          <string notr="true">
QPushButton {
    border-radius: 8px; 
    background-color: rgb(170, 170, 255);
    color: rgb(0, 0, 0);
    padding: 6px 12px;
    border: none; 
}

QPushButton:pressed {
    background-color: ;
	color: rgb(151, 151, 227);
}
</string>
         </property>
         <property name="text">
          <string>Рассчитать интерполяцию сплайнами</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="resultsGroup">
         <property name="styleSheet">
          <string notr="true">color: rgb(170, 170, 255);</string>
         </property>
         <property name="title">
          <string>Результаты - Максимальные погрешности</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTextEdit" name="resultsText">
            <property name="styleSheet">
             <string notr="true">background-color: rgb(206, 168, 218);
color: rgb(0, 0, 0);</string>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="linearSplineGroup">
      <property name="styleSheet">
       <string notr="true">color: rgb(170, 170, 255);</string>
      </property>
      <property name="title">
       <string>Линейный сплайн S1(x)</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QWidget" name="linearPlotWidget" native="true"/>
       </item>
       <item>
        <widget class="QPushButton" name="linearToggleBtn">
         <property name="styleSheet">
          <string notr="true">
QPushButton {
    border-radius: 8px; 
    background-color: rgb(170, 200, 255);
    color: rgb(0, 0, 0);
    padding: 6px 12px;
    border: none; 
}

QPushButton:pressed {
    background-color: ;
	color: rgb(151, 181, 227);
}
</string>
         </property>
         <property name="text">
          <string>Показать погрешность</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="parabolicSplineGroup">
      <property name="styleSheet">
       <string notr="true">color: rgb(170, 170, 255);</string>
      </property>
      <property name="title">
       <string>Параболический сплайн S2(x)</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QWidget" name="parabolicPlotWidget" native="true"/>
       </item>
       <item>
        <widget class="QPushButton" name="parabolicToggleBtn">
         <property name="styleSheet">
          <string notr="true">
QPushButton {
    border-radius: 8px; 
    background-color: rgb(170, 255, 200);
    color: rgb(0, 0, 0);
    padding: 6px 12px;
    border: none; 
}

QPushButton:pressed {
    background-color: ;
	color: rgb(151, 227, 181);
}
</string>
         </property>
         <property name="text">
          <string>Показать погрешность</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="cubicSplineGroup">
      <property name="styleSheet">
       <string notr="true">color: rgb(170, 170, 255);</string>
      </property>
      <property name="title">
       <string>Кубический сплайн S3(x)</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QWidget" name="cubicPlotWidget" native="true"/>
       </item>
       <item>
        <widget class="QPushButton" name="cubicToggleBtn">
         <property name="styleSheet">
          <string notr="true">
QPushButton {
    border-radius: 8px; 
    background-color: rgb(255, 200, 170);
    color: rgb(0, 0, 0);
    padding: 6px 12px;
    border: none; 
}

QPushButton:pressed {
    background-color: ;
	color: rgb(227, 181, 151);
}
</string>
         </property>
         <property name="text">
          <string>Показать погрешность</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>